### PR TITLE
feat: Connect components in React Native [part 2]

### PR DESCRIPTION
**Pull request by joaquim-verges**
Connected button and modal implementation in RN

---

## PR-Codex overview
The focus of this PR is to refactor and optimize the codebase by moving and updating imports, modifying function parameters, and improving file organization.

### Detailed summary
- Updated `wallets` property in `useAutoConnect.ts`
- Moved `TokenInfo` import in various files
- Updated import paths for `socialIcons`
- Refactored `useSendToken` in web and native
- Updated `defaultTokens` import paths

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/core/hooks/connection/ConnectButtonProps.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/TokenSelector.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/ViewFunds.tsx`, `packages/thirdweb/src/react/web/ui/components/WalletImage.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/PayWithCrypto.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/ConnectButton.tsx`, `packages/thirdweb/src/react/native/ui/connect/ErrorView.tsx`, `packages/thirdweb/src/react/native/ui/connect/SuccessView.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/BuyScreen.tsx`, `packages/thirdweb/src/react/native/ui/components/ChainIcon.tsx`, `packages/react-native-adapter/README.md`, `packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Details.tsx`, `packages/thirdweb/src/react/web/hooks/wallets/useAutoConnect.ts`, `packages/thirdweb/src/react/native/ui/components/RNImage.tsx`, `packages/thirdweb/src/react/native/ui/components/Address.tsx`, `packages/thirdweb/src/react/native/ui/components/input.tsx`, `packages/thirdweb/src/react/native/ui/connect/InAppWalletUI.tsx`, `packages/thirdweb/src/react/native/ui/components/Header.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/defaultTokens.ts`, `packages/thirdweb/src/react/native/ui/connect/ExternalWalletsList.tsx`, `packages/thirdweb/src/react/native/ui/components/WalletImage.tsx`, `packages/thirdweb/src/react/native/ui/connect/ReceiveScreen.tsx`, `packages/thirdweb/src/react/core/hooks/wallets/useSendToken.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/SendFunds.tsx`, `packages/thirdweb/package.json`, `packages/thirdweb/src/react/web/ui/components/ChainIcon.tsx`, `packages/thirdweb/src/react/native/ui/connect/TokenListScreen.tsx`, `packages/thirdweb/src/react/native/ui/connect/ConnectButton.tsx`, `packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx`, `packages/thirdweb/src/react/native/ui/connect/SendScreen.tsx`, `packages/thirdweb/src/react/native/ui/connect/ConnectModal.tsx`, `packages/thirdweb/src/react/native/ui/icons/svgs.ts`, `packages/thirdweb/src/react/native/ui/connect/ConnectedModal.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/icons/dataUris.ts`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3438?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3438** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3438?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3246** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3246?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joaquim-verges and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/react/native/ui/components/Address.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export const Address = ({ account, theme, addressOrENS }: AddressProps) => {
+  const [copySuccess, setCopySuccess] = useState(false);
+  useEffect(() => {
+    if (copySuccess) {
+      setTimeout(() => {
+        setCopySuccess(false);
+      }, 2500);
+    }
+  }, [copySuccess]);
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/components/Address.tsx:25]: have to clean up this timeout if it was set in the cleanup function

File: packages/thirdweb/src/react/native/ui/components/Address.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <TouchableOpacity
+      style={styles.addressContainer}
+      onPress={() => {
+        // @ts-ignore - missing native types
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/components/Address.tsx:30]: prefer `@ts-expect-error`

```suggestion
        // @ts-expect-error - missing native types
```

File: packages/thirdweb/src/react/native/ui/components/WalletImage.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    queryKey: [
+      "wallet-image-or-ens",
+      wallet.id,
+      wallet.getAccount()?.address,
+      ensAvatar,
+    ],
+    queryFn: async (): Promise<string> => {
+      if (ensAvatar) {
+        return ensAvatar;
+      }
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/components/WalletImage.tsx:33]: just keep this entirely outside of the useQuery, it doesn't need to be in here

File: packages/thirdweb/src/react/native/ui/components/WalletImage.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  size: number;
+  ensAvatar?: string | null;
+}) => {
+  const { wallet, ensAvatar, size } = props;
+  const { data: imageData } = useQuery({
+    queryKey: [
+      "wallet-image-or-ens",
+      wallet.id,
+      wallet.getAccount()?.address,
+      ensAvatar,
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/components/WalletImage.tsx:28]: remove

File: packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx [hunk 5]
```diff
@@ -41,135 +28,51 @@ export function ConnectedButton(
...
+            type="subtext"
+            style={{
+              color: theme.colors.secondaryText,
+              fontSize: fontSize.sm,
+            }}
+          >
+            {balanceQuery.data
+              ? Number(balanceQuery.data.displayValue).toFixed(3)
+              : "---"}{" "}
+            {balanceQuery.data?.symbol}{" "}
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx:68]: why the ending space? `{" "}`

File: packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx [hunk 6]
```diff
@@ -41,135 +28,51 @@ export function ConnectedButton(
...
+            theme={theme}
+            type="subtext"
+            style={{
+              color: theme.colors.secondaryText,
+              fontSize: fontSize.sm,
+            }}
+          >
+            {balanceQuery.data
+              ? Number(balanceQuery.data.displayValue).toFixed(3)
+              : "---"}{" "}
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx:67]: is the `---` the "loading" state?

File: packages/thirdweb/src/react/native/ui/connect/TokenListScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+};
+
+export const TokenListScreen = (props: TokenListScreenProps) => {
+  const supportedTokens = props.supportedTokens || defaultTokens;
+  const chain = useActiveWalletChain();
+  const account = useActiveAccount();
+  const tokens = chain ? supportedTokens[chain.id] || [] : [];
+
+  return (
+    <>
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/connect/TokenListScreen.tsx:33]: empty fragment?

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Pull request merged**